lib/readline.js

```diff
@@ -556,28 +556,40 @@ Interface.prototype._line = function() {
 
 
 Interface.prototype._historyNext = function() {
-  if (this.historyIndex > 0) {
-    this.historyIndex--;
-    this.line = this.history[this.historyIndex];
-    this.cursor = this.line.length; // set cursor to end of line.
-    this._refreshLine();
+  // Set historyIndex to the latest item in the history when repl loads
+  // and this is the first action that is invoked
+  if (this.historyIndex === null) this.historyIndex = this.history.length - 1;
 
-  } else if (this.historyIndex === 0) {
-    this.historyIndex = -1;
+  if ((this.historyIndex + 1) === this.history.length) {
     this.cursor = 0;
     this.line = '';
     this._refreshLine();
+  } else if (this.historyIndex < this.history.length) {
+    this.historyIndex++;
+    this.line = this.history[this.historyIndex];
+    this.cursor = this.line.length;
+    this._refreshLine();
   }
 };
 
 
 Interface.prototype._historyPrev = function() {
-  if (this.historyIndex + 1 < this.history.length) {
-    this.historyIndex++;
-    this.line = this.history[this.historyIndex];
-    this.cursor = this.line.length; // set cursor to end of line.
-
-    this._refreshLine();
+  // Set historyIndex to the latest item in the history when repl loads
+  // and this is the first action that is invoked
+  if (this.historyIndex === null) this.historyIndex = this.history.length - 1;
+
+  if (this.history.length) {
+    if (this.historyIndex > 0) {
+      this.line = this.history[this.historyIndex];
+      this.cursor = this.line.length; // set cursor to end of line
+      this.historyIndex--;
+      this._refreshLine();
+    }
+    else {
+      this.line = this.history[0];
+      this.cursor = this.line ? this.line.length : '';
+      this._refreshLine();
+    }
```

Comment:
I suggest not changing `readline` at all, the logic should work with just the history file storage being reversed. That is a very large change, probably suitable for another PR. Marking as `semver-major` for now due to this.